### ISSUE TITLE
Turning on asset aggregation should not be encouraged

### ISSUE THREAD

**Issue by marfillaster** [pantheon-systems/documentation#6202]
Re: [Drupal Performance and Caching Settings](https://pantheon.io/docs/drupal-cache)

Priority: Low/Medium/High (choose one, remove the other options)

## Issue Description:

HTTP/2 allows for multiplexing which lets multiple requests share a single TCP connection, allowing several assets to download in parallel without the unnecessary overhead of establishing multiple connections[[1](https://blog.cloudflare.com/http-2-for-web-developers/#:~:text=Multiplexing%20lets%20multiple%20requests%20share%20a%20single%20TCP%20connection,%20allowing%20several%20assets%20to%20download%20in%20parallel%20without%20the%20unnecessary%20overhead%20of%20establishing%20multiple%20connections)]. This new feature removes the need to for concatenating/inlining of JavaScript and CSS. 

Advagg, a popular contrib module, extends Drupal built-in asset aggregation feature. However, the module can potentially generate a ton of aggregated files which can impact the performance of the persistent filesystem in the platform. Customers should be advised to opt-out of aggregation the site when possible.

**Comment by alexfornuto** [comment]
Thanks for the issue @marfillaster! Can you cite any existing documentation on best practices re: multiplexing vs aggregation?

**Comment by marfillaster** [comment]
- https://developers.google.com/web/fundamentals/performance/http2#request_and_response_multiplexing
-  https://blog.cloudflare.com/http-2-for-web-developers/#:~:text=Multiplexing%20lets%20multiple%20requests%20share%20a%20single%20TCP%20connection,%20allowing%20several%20assets%20to%20download%20in%20parallel%20without%20the%20unnecessary%20overhead%20of%20establishing%20multiple%20connections
- https://css-tricks.com/musings-on-http2-and-bundling/
-https://community.akamai.com/customers/s/article/Effect-of-HTTP-2-on-Current-Web-Performance-Optimizations-Techniques?language=en_US

**Comment by alexfornuto** [comment]
Thanks for the context @marfillaster! We're going to seek out a Drupal SME to add context from a Drupal 7 perspective.

Reference: https://www.drupal.org/forum/general/general-discussion/2019-11-12/http2-equals-no-aggregation

**Comment by wordsmither** [comment]
Potentially could be added to Modules w/Known Issues page.

**Comment by wordsmither** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Turning on asset aggregation should not be encouraged #6202](https://trello.com/c/de98sW2L/40-turning-on-asset-aggregation-should-not-be-encouraged-6202)

**Comment by wordsmither** [comment]
@kporras07 If you're not the right person to ask, please let me know.  Would you agree that the issue described should be added to our [Modules w/Known Issues](https://pantheon.io/docs/modules-known-issues) page?

**Comment by kporras07** [comment]
@wordsmither I would rephrase it a bit but still keep the recommendation in the same page.

Previously:
'On the Live environment, make sure to enable "Aggregate and compress CSS files" and "Aggregate and compress JavaScript files". This is critical for page render times by reducing the number of HTTP requests and reducing the amount of data transferred.'

Proposed:
'Pantheon uses HTTP/2 which _may_ make aggregation not needed for your site by allowing files to download in parallel. However, this will depend on factors like files size, files amount, your end users' browsers and some other stuff. If you aren't familiar with how those factors apply to your site, we recommend you to keep the aggregation enabled.'
